### ISSUE TITLE
Block Navigation menu enabled even when there are no blocks

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#12040]
When a post doesn't have blocks yet, for example because it's a newly created post, the Block Navigation menu is enabled even though there are no blocks to list. Clicking the button shows a popover with the message "No blocks created yet." which is a bit pointless.

<img width="426" alt="screenshot 2018-11-18 at 18 24 06" src="https://user-images.githubusercontent.com/1682452/48675877-33e70a00-eb5f-11e8-8255-16238e256503.png">

Not sure this is the more consistent user experience as, for example, the "Content structure" tool which is there in the UI, close to the Block Navigation menu, has actually the opposite behavior: it is disabled because there are no headings or content to analyze yet.

Checking the semantics and interaction consistency of all the buttons in this part of the UI, seems there are at least 3 different initial behaviors that are potentially confusing for users:

Undo and Redo buttons:
  - they're not really disabled, they use an `aria-disabled="true"` attribute
  - they're still focusable
  - their tooltip is visible

As discussed in previous issues / PRs, they're not disabled to avoid a focus loss when using a keyboard and there are no more undo / redo steps.

Content structure button:
  - it uses a `disabled` attribute, so it's really disabled
  - it's not focusable
  - the tooltip is not visible

As it's not focusable and its tooltip is not visible, the discoverability of this tool is a bit reduced.

Block Navigation button:
  - it's always enabled
  - it's always focusable 
  - the tooltip is visible
  - its popover is always rendered even when it's empty because there are no blocks inserted yet, which is a noticeable different behavior compared to the Content structure tool

From an accessibility perspective, using `aria-disabled="true"` and "noop'ing" an UI control is equivalent to disabling it with a `disable` attribute, as long as it is perceived as disabled by all users, i.e. also visually. The relevant difference is that a control with `aria-disabled="true"` is still focusable. Removing focusability from disabled elements can offer users both advantages and disadvantages. A must read recommendation can be found in the ARIA authoring practices:

5.7 Focusability of disabled controls
https://www.w3.org/TR/wai-aria-practices-1.1/#kbd_disabled_controls

Although those recommendations apply to ARIA widgets, the Gutenberg top bar can be considered a "toolbar", where the ability to discover the functionalities within the toolbar is a primary concern. I'd recommend to adopt a consistent convention for the focusability of these controls and always use `aria-disabled="true"` / "noop". The Block navigation menu should not open its panel when there are no blocks.

Lastly, minor but important for consistency: sometimes the tooltips use sentence case, sometimes title case:

<img width="250" alt="screenshot 2018-11-18 at 18 15 46" src="https://user-images.githubusercontent.com/1682452/48675863-d94dae00-eb5e-11e8-9373-d15a2812caf3.png">
<img width="268" alt="screenshot 2018-11-18 at 18 15 56" src="https://user-images.githubusercontent.com/1682452/48675864-d94dae00-eb5e-11e8-9a24-fc869b7c14bb.png">
<img width="260" alt="screenshot 2018-11-18 at 18 16 05" src="https://user-images.githubusercontent.com/1682452/48675865-d9e64480-eb5e-11e8-9ece-c4f611a73009.png">
